[Docs] Dynamic Layout content populated with cpp

mythrz opened the pull request:
- Dynamic Layout content populated with cpp
- the example expects another ui component like the dropdown to already be present.

## Change summary

Part of https://github.com/o3de/o3de.org/issues/2705

- Covers how to in how to include and inherit the bus
- how to override the functions/methods/events so it compiles without errors
- how to clone/copy the existing option so it matches the new content size

### Submission Checklist:

* [x ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ x] **Help the user** - Does the documentation show the user something *meaningful*?

mythrz committed:
- Dynamic Layout content populated with cpp
- the example expect another ui component like the dropdown to already be present.

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update dynamic-layout.md

markdown with cpp so code is not only white on black

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update dynamic-layout.md

link for lua example. This is part of the LyShineExamples

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update dynamic-layout.md

Code sample includes how to clone and then modify the text-option using a vector of strings.
Extracted the code from here, and adapted it to the needs of the example:
https://github.com/o3de/o3de/blob/f17b405699c55400a985b8741af2649c39f11315/Gems/LyShineExamples/Code/Source/UiTestScrollBoxDataProviderComponent.cpp#L61

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update dynamic-layout.md

Included a link to another page, UIInitializationBus.
The dynamic layout should always be populated after ALL necessary UI elements are loaded, otherwise it can lead to unexpected results. The link at the bottom of the page, will suggest/guide the user to continue the tutorial with the use of UiButtonNotificationBus

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

ShaunaGordon commented on content/docs/user-guide/interactivity/user-interface/components/layout/dynamic-layout.md:
```diff
@@ -32,3 +32,103 @@ For a [**LayoutGrid**](components-layout-grid) element, the cell size of the **L
+
+**Cpp**
+
+To create your own UI components, check the [Working with UI Components](https://www.docs.o3de.org/docs/user-guide/interactivity/user-interface/components/working/) page.
```
Internal links should be relative, not absolute.

mythrz committed:
Update dynamic-layout.md

fix for the internal urls, with relative path.

Kept the absolute github link thought. Nothing to do here.
https://github.com/o3de/o3de/blob/f17b405699c55400a985b8741af2649c39f11315/Gems/LyShineExamples/Assets/UI/Scripts/LyShineExamples/Dropdown/MultiSelectionDropdown.lua#L29

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

ShaunaGordon commented on content/docs/user-guide/interactivity/user-interface/components/layout/dynamic-layout.md:
```diff
@@ -32,3 +32,103 @@ For a [**LayoutGrid**](components-layout-grid) element, the cell size of the **L
+
+If you use the vector of strings that contains 3 indexes it will modify the text-element of each option and populate the dropdown as expected.
+
+<img width="236" height="157" alt="image" src="https://github.com/user-attachments/assets/07a1ac9b-dd90-4e17-b6d7-8754f114e30a" />
```
Is there a particular reason this is using github's attachment cdn? It would be better if the image is stored in the project and referenced interally, unless there's a compelling reason to do otherwise.

mythrz replied:
Probably copy-past it instead of adding the file. No particular reason. Tired I guess (:
Created a new PR to add the file: https://github.com/o3de/o3de.org/pull/2731

Changing to this then. sec
![alt text](image2.png)

mythrz committed:
Merge branch 'development' into patch-4

mythrz committed:
Update dynamic-layout.md

Aprove 2731 one first: https://github.com/o3de/o3de.org/pull/2731
It will include the missing image
Then this one, that references it from the source.

Or manually edit this PR...

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>
